nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -708,32 +819,16 @@ def setup_training_dataloader(self):
 
     def setup_eval_dataloader(self, datasets, data_cfg):
         dataloaders = []
+        if not isinstance(datasets, list):
+            return self.build_data_loader(dataset=datasets, data_cfg=data_cfg, consumed_samples=0,)
         for dataset in datasets:
             eval_dl = self.build_data_loader(dataset=dataset, data_cfg=data_cfg, consumed_samples=0,)
             dataloaders.append(eval_dl)
         return dataloaders
 
-    def _reset_activation_checkpointing_args(self):
```

Comment:
@stevehuang52 do you remember whether and why this change is necessary?

Reply:
it's from the source PR https://github.com/NVIDIA/NeMo/pull/6915